### PR TITLE
Feature/ACRDDF-73 Adapts to Java Spring templates

### PR DESCRIPTION
**Pull request by rafaellogos**
Adapts the generate-it to be able to generate Java templates by:
- Including an option to ignore custom folders
- Including an option to copy files without parsing

### CODE REVIEW

**Comment by p-mcgowan** [comment]
@j-d-carmichael 
we're aiming to make some changes to enable opt-out of node-specific generation (eg prettier on rendered files) so that we can, in theory, use this for other languages.

I'm going to run some projects through this merge to make sure everything still goes through, but would like to make sure nothing on your side changes before approving

**Comment by p-mcgowan** [comment]
agreed - we should definitely update the docs to reflect any changes and features, and I think it would be good to make sure options from cli are available as nodegenrc options too.

Will look to see if the parts of the readme that are changing need any other polishes, and could definitely put it on the radar to do another docs pass to make sure everything is still up to date as I think I recall there being a few places where one or two things were not super clear, but would suggest doing a more comprehensive pass on that in a docs feature.

**Comment by p-mcgowan** [comment]
rabbit hole found....

I went through, changed many links, altered a few minor things in the docs. 

I also added "caching" behaviour to the isFileToIgnore method since re-creating the same regex each file is not necessary.

[generate-it-java.patch.zip](https://github.com/acr-lfr/generate-it/files/8592265/generate-it-java.patch.zip)

File: docs/_pages/features.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ It is required that the 1st run is done with an internet connection, this flag w
 ## Stub (domain) versions
 The core will store the last few versions of stub files generated in a cache folder to compare future changes:
 ```
-.openapi-nodegen/cache/compare
+.generate-it/cache/compare
```
Comment (p-mcgowan) [inline docs/_pages/features.md:43]: whoops - I did a find and replace fail here. This should still be `.openapi-nodegen`

Comment (acrontum-silva) [reply docs/_pages/features.md]: Fixed!

Comment (rafaellogos) [reply docs/_pages/features.md]: Fixed

**Comment by p-mcgowan** [comment]
looks good from my side, @j-d-carmichael anything else from yours?

**Review by p-mcgowan — APPROVED** [review]

**Comment by j-d-carmichael** [comment]
Look great to me! Awesome to see those docs get a little love, it has been a little while!

**Pull request merged**